common/notify_stream/src/lib.rs

```diff
@@ -176,9 +184,14 @@ impl Watcher {
                     EventKind::Modify(ModifyKind::Data(DataChange::Any)) => {
                         Some(Event::Write(received.paths.first().unwrap().clone()))
                     }
-                    EventKind::Modify(ModifyKind::Name(RenameMode::From)) => {
-                        Some(Event::Remove(received.paths.first().unwrap().clone()))
-                    }
+                    EventKind::Modify(ModifyKind::Name(RenameMode::From)) => received
```

Comment:
Do you think there's value in adding this type of catch to the other EventKinds? Or, is this type of thing just specific to the notify-rs behavior on this type of event?

Reply:
It is abnormal behavior of notify-rs. This workaround is about "Rename From" event only. It is part of event pair: "From" + "To". Bad case happens when notify-rs did not find tracked path in its own map at the time it composes "From" event. We should not ignore empty path cases in other events.

Reply:
implemented